lib/miq_automation_engine/engine/miq_ae_service.rb

```diff
@@ -52,6 +59,14 @@ def initialize(ws)
       self.class.add(self)
     end
 
+    def enable_rbac
+      Thread.current.thread_variable_set(:miq_rbac, true)
+    end
```

Comment:
could this take a block instead?

Need to be very aware what the values of `nil`, `true`, and `false` mean.
My code example assumes `rbac` is enabled by default. code accordingly

```ruby
def disable_rbac
  old_miq_rbac = Thread.current.thread_variable_get(:miq_rbac)
  Thread.current.thread_variable_set(:miq_rbac, false) unless old_miq_rbac == false
  yield
ensure
  Thread.current.thread_variable_set(:miq_rbac, true) unless old_miq_rbac == false
end
```